src/manager/msWorkspaceManager.ts

```diff
@@ -735,7 +740,51 @@ export class MsWorkspaceManager extends MsManager {
         if (!msWorkspace || !metaWindow.msWindow) {
             return;
         }
-        this.setWindowToMsWorkspace(metaWindow.msWindow, msWorkspace);
+        this.setWindowToMsWorkspaceWithCreationChaosProtection(
+            metaWindow.msWindow,
+            msWorkspace
+        );
+    }
+
+    /**
+     * On creation window can jump from monitor to monitor quickly for obscure reason until they stabilize on the correct one
+     * In order to get ride of the visual glitch of this behavior we guard rapid changes
+     */
+    setWindowToMsWorkspaceWithCreationChaosProtection(
```

Comment:
Ugh, this is ugly, bug I suppose

Reply:
Some gnome apps like the gnome terminal does jump around monitor on their creation causing tiling flickering I didn't found a better way to handle this